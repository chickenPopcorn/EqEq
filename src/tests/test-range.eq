SomeCtx = {
  a = { 42; }
  b = { 32; }

}

SomeCtx:find a {
   print("print for simple find\n");
   print("a = %0.0f\n", a);
   print("b = %0.0f\n", b);
}

SomeCtx:find a with c in range(0,3); {
  print("print for find with range (0,3)\n");
  print("a = %0.0f\n", a);
  print("b = %0.0f\n", b);
  print("c = %0.0f\n", c);
}

/*
SomeCtx:find a with c in range(2,-1); {
  print("print for find with range (2,-1)\n");
  print("a = %0.0f\n", a);
  print("b = %0.0f\n", b);
  print("c = %0.0f\n", c);
}


SomeCtx:find a with c in range(-1,-1); {
  print("print for find with range (-1,-1)\n");
  print("a = %0.0f\n", a);
  print("b = %0.0f\n", b);
  print("c = %0.0f\n", c);
}

SomeCtx:find a with b = 10; c in range(0,2); {
  print("print for find with range (0,2)\n");
  print("a = %0.0f\n", a);
  print("b = %0.0f\n", b);
  print("c = %0.0f\n", c);
}
*/