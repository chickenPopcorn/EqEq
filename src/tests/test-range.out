print for simple find
a = 42
b = 32
print for find with range (0,3,1)
a = 42
b = 32
c = 0
print for find with range (0,3,1)
a = 42
b = 32
c = 1
print for find with range (0,3,1)
a = 42
b = 32
c = 2
print for find with range (0,3,1)
a = 42
b = 32
c = 3
print for find with range (2,-1,-1)
a = 42
b = 32
c = 2
print for find with range (2,-1,-1)
a = 42
b = 32
c = 1
print for find with range (2,-1,-1)
a = 42
b = 32
c = 0
print for find with range (2,-1,-1)
a = 42
b = 32
c = -1
print for find with range (-1,-1,1)
a = 42
b = 32
c = -1
print for find with range (0,4,2)
a = 42
b = 10
c = 0
print for find with range (0,4,2)
a = 42
b = 10
c = 2
print for find with range (0,4,2)
a = 42
b = 10
c = 4
print for find with range (0,2)
a = 42
b = 10
c = 0
print for find with range (0,2)
a = 42
b = 10
c = 1
print for find with range (0,2)
a = 42
b = 10
c = 2
print for find with range (2)
a = 42
b = 10
c = 0
print for find with range (2)
a = 42
b = 10
c = 1
print for find with range (2)
a = 42
b = 10
c = 2
